### ISSUE TITLE
Fatal on error `has or is using private name` on an already export-ed Class

### ISSUE THREAD

**Issue by zixia** [palantir/tslint#3173]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.4.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
import {
  widget,
  Widgets,
}             from 'blessed'

declare module 'blessed' {
  namespace widget {
    class Image extends Widgets.ImageElement {}
  }

  namespace Widgets {
    interface Border {
      top?:    boolean,
      bottom?: boolean,
    }
  }
}
```

with `tslint.json` configuration:

```shell
tslint --project tsconfig.json --type-check 
```

#### Actual behavior

> [ts] 'extends' clause of exported class 'Image' has or is using private name 'Widgets'.

However, `ts-node` & `tsc` can run the code without any problem.

#### Expected behavior

Should not show `private name `Widgets` because it's already `export`-ed.

**Comment by ajafff** [comment]
The quick fix for your problem is omitting `--type-check`. It's no longer needed to use type checked rules.
You will then no longer see any compile errors in the lint output.

**Comment by zixia** [comment]
Yes, that's what I did at last.

But what do you mean "it's no longer needed", had `--type-check` deprecated?

> It's no longer needed to use type checked rules.

**Comment by ajafff** [comment]
`--type-check` is not deprecated yet. It's just no longer needed for typed rules to work, `--project` is sufficient for that.

It probably will be deprecated. It's no longer useful and we frequently get bug reports about it. I don't think it's worth keeping it around.

**Comment by zixia** [comment]
It's a good news that `--project` is sufficient for checking.

Thanks for the explanation!

**Comment by trusktr** [comment]
I've opened a request to fix issues like these in TypeScript by bringing declaration files to parity with language features. https://github.com/microsoft/TypeScript/issues/35822
